janusgraph-test/src/main/java/org/janusgraph/blueprints/AbstractJanusGraphProvider.java

```diff
@@ -70,38 +73,46 @@
  */
 public abstract class AbstractJanusGraphProvider extends AbstractGraphProvider {
 
-    private static final Set<Class> IMPLEMENTATION = new HashSet<Class>() {{
-        add(StandardJanusGraph.class);
-        add(StandardJanusGraphTx.class);
-
-        add(StandardVertex.class);
-        add(CacheVertex.class);
-        add(PreloadedVertex.class);
-        add(EdgeLabelVertex.class);
-        add(PropertyKeyVertex.class);
-        add(VertexLabelVertex.class);
-        add(JanusGraphSchemaVertex.class);
-        add(EmptyVertex.class);
-
-        add(StandardEdge.class);
-        add(CacheEdge.class);
-        add(EdgeLabel.class);
-        add(EdgeLabelVertex.class);
-
-        add(StandardVertexProperty.class);
-        add(CacheVertexProperty.class);
-        add(SimpleJanusGraphProperty.class);
-        add(CacheVertexProperty.class);
-        add(FulgoraVertexProperty.class);
-
-        add(JanusGraphVariables.class);
-
-        add(FulgoraElementTraversal.class);
-    }};
+	private enum Implementation {
+		INSTANCE;
+		private final Set<Class> classes;
+		private Implementation() {
+			classes = ImmutableSet.of(StandardJanusGraph.class,
```

Comment:
@pluradj ill visit this PR later this week, I got distracted by figuring out how to address our 40 hour release build time...